WooCommerce/Classes/POS/Presentation/Reusable Views/WavesProgressViewStyle.swift

```diff
@@ -0,0 +1,96 @@
+import SwiftUI
+
+struct WaveProgressViewStyle: ProgressViewStyle {
+    func makeBody(configuration: Configuration) -> some View {
+        WaveProgressView(configuration: configuration)
+    }
+}
+
+private struct WaveProgressView: View {
+    let configuration: ProgressViewStyleConfiguration
+    private let animationDuration: Double = 0.5
+    private let radii: [CGFloat] = [40.0, 70.0, 100.0]
+    private let inactiveInset: CGFloat = 0.025
+
+    @State private var activeArcIndex: Int = 0
+
+    private var arcCount: Int {
+        radii.count
+    }
+
+    private var xOffset: CGFloat {
+        guard let lastRadius = radii.last,
+              let firstRadius = radii.first,
+              lastRadius > firstRadius else {
+            return 0
+        }
+        let centreOfWaves = (lastRadius - firstRadius) / 2
+        return -((firstRadius + centreOfWaves) / 2)
+    }
+
+    var body: some View {
+        ZStack {
+            ForEach(0..<arcCount, id: \.self) { index in
+                let isActive = index == activeArcIndex
+                let radius = radii[index]
+
+                ArcShape(startAngle: .degrees(333),
+                         endAngle: .degrees(27))
+                .trim(from: isActive ? 0.0 : 0 + inactiveInset,
+                      to: isActive ? 1.0 : 1 - inactiveInset)
+                .stroke(isActive ? Color(.wooCommercePurple(.shade60)) : Color(.wooCommercePurple(.shade40)),
+                        lineWidth: isActive ? 10 : 7)
+                .frame(width: radius, height: radius)
+                .animation(.easeInOut(duration: animationDuration),
+                           value: activeArcIndex)
+            }
+        }
+        .offset(x: xOffset)
+        .frame(width: 130, height: 115)
```

Comment:
I think it would be good to move some of the constants that are configurable to a `Constants` enum or default properties for easier configuration.

Reply:
I considered it, but I didn't want to imply that it was actually set up for that customisation yet.

I took the approach of showing the numbers in line where they're 1-use only and we've not yet considered configuration, and using a var where there's actually a relationship between the two numbers.

If we use it elsewhere, we can make it configurable as we go